src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -255,6 +257,49 @@ runtime::gpu::GPU_ExternalFunction::~GPU_ExternalFunction()
 {
 }
 
+void runtime::gpu::GPU_ExternalFunction::optimize_and_assemble()
```

Comment:
https://en.wikipedia.org/wiki/Single_responsibility_principle

Reply:
Fair point, it was done so that the GPUAllocator would correctly go out of scope when the reservation requests had expired (after the pool reservation and all kernel allocations were made). Instead, I added GPUAllocator::close() to force the closure of the allocator so the device allocations can proceed without the requiring GPUAllocator to explicitly deconstruct.